Rakefile

```diff
@@ -10,13 +13,76 @@ Dir.glob('**/*.rake').each do |task_file|
   load task_file
 end
 
-Rake::TestTask.new do |t|
-  t.libs << 'test'
-  t.pattern = 'test/**/*_test.rb'
-  t.warning = false
+RuboCop::RakeTask.new
+
+task default: :spec
+task 'release:test' => :spec
+
+PRIVATE_KEY = 'spec/fixtures/pem/privatekey.pem'
+CERTIFICATE = 'spec/fixtures/pem/certificate.pem'
```

Comment:
Is it possible we can generate these on the fly for when we run the tests? AWS Security does not like embedded / fixtures for certificates and keys.

Reply:
I didn't write this code, not sure, can look into it one day

Reply:
Perhaps @bobf can help or provide insight on why this was needed.

Reply:
Okay...so these update the fixtures in json with the proper values based on the content. (so if you change content, must run the rake task to update). 

It looks like the signing cert comes from AWS SNS publicly and it's used to do that. Pulling live seems like it'd be fragile/slower.

Reply:
Is there a public link for that? I'd rather get it from the public source even if it's slower. It shouldn't be fragile I don't think? We could always fetch on each run.

Reply:
Ugh, it seems to be, but I don't know what the current one is. You can see how it refs it in the json fixtures themselves—like it returns a `SigningCertURL`

Reply:
I'd say fragile as it has like a UUID on the URL. 

https://docs.aws.amazon.com/sns/latest/dg/sns-verify-signature-of-message.html

Okay, here's what it says

Reply:
```
Get the X509 certificate that Amazon SNS used to sign the message. The SigningCertURL value points to the location of the X509 certificate used to create the digital signature for the message. Retrieve the certificate from this location.
Extract the public key from the certificate. The public key from the certificate specified by SigningCertURL is used to verify the authenticity and integrity of the message.
```

I'm not sure this a concern. I think the cert is like arbitrary?

Reply:
Like we are stubbing out the call to get the pem with the one stored in the app.

Reply:
I'll investigate this and get back to you.

Reply:
I believe whats being done here is fairly reasonable since we are generating random test certificates and then updating the signatures in the fixtures to match.

However - It seems like this is something that could be done instead as an rspec before(:all) and then avoids any need for running these tasks when tests are changed or storing test certs.

Reply:
Hmm. Well, it's saving changes to the fixture files. Or are you saying we don't save the files, like merge it into the data returned? 

Seems a little more complicated